core/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -172,6 +172,9 @@ public ParserContext createMultiFieldContext(ParserContext in) {
     private final String simpleName;
 
     public Mapper(String simpleName) {
+        if (Strings.isNullOrEmpty(simpleName)) {
+            throw new IllegalArgumentException("name cannot be null and empty string");
+        }
```

Comment:
I would distinguish the null check from the empty check, as a null simpleName means that we have a bug while an empty simpleName means the issue is on the user end. So maybe something like this:
```
Objects.requireNonNull(simpleName);
if (simpleName.isEmpty()) {
    throw new IllegalArgumentException("name cannot be an empty string");
}
```